internal/repository/websocket.go

```diff
@@ -60,23 +61,32 @@ func NewWebSocket() (*WebSocket, error) {
 	}, nil
 }
 
-func (w *WebSocket) sendMessageToRetro(ctx context.Context, message types.WebSocketMessage) error {
-	retrospectiveID, ok := ctx.Value("retrospective_id").(uuid.UUID)
-	if !ok {
-		return fmt.Errorf("retrospective id not found")
+func (w *WebSocket) sendMessageToRetro(ctx context.Context, message types.WebSocketMessage, retrospectiveID *uuid.UUID) error {
+	if retrospectiveID == nil {
+		id, ok := ctx.Value("retrospective_id").(uuid.UUID)
+		if !ok {
+			return fmt.Errorf("retrospective id not found")
+		}
+		retrospectiveID = &id
 	}
 
-	connections := w.connections[retrospectiveID]
+	connections := w.connections[*retrospectiveID]
 	if connections == nil {
 		return nil
 	}
 
-	for _, conn := range connections {
+	for i, conn := range connections {
+		if conn == nil {
+			continue
+		}
 		err := conn.WriteJSON(message)
 		if err != nil {
 			log.Printf("Error sending message %+v to connection: %v", message, err)
+			conn.Close()
```

Comment:
I Think this can cause a zombie connection in the front.

If the backend connection go through any problem, and the send message fails, it will try to close the connection, but we can't send messages 'cause were offline. The connection is `nilled` in back, but for front we still connected.

We do not have any heartbeat or ping system, so frontend thinks we're connected, and keep connected waiting for messages